File: text-to-speech/quickstart/quickstart.py
import os
import sys

import requests

api_key = os.environ.get("SMALLEST_API_KEY")
if not api_key:
    print("Error: set SMALLEST_API_KEY environment variable")
    print("Get your key at https://app.smallest.ai/dashboard/settings/apikeys")
    sys.exit(1)

response = requests.post(
    "https://api.smallest.ai/waves/v1/lightning-v3.1/get_speech",
    headers={
        "Authorization": f"Bearer {api_key}",
        "Content-Type": "application/json",
    },
    json={
        "text": "Hello! Welcome to Smallest AI. This is your first text-to-speech generation.",
        "voice_id": "sophia",
        "sample_rate": 24000,
        "output_format": "wav",
    },
)

if response.status_code != 200:
    print(f"Error ({response.status_code}): {response.text}")
    sys.exit(1)

with open("output.wav", "wb") as f:
    f.write(response.content)
print(f"Done! Saved output.wav ({len(response.content):,} bytes)")

File: text-to-speech/quickstart/quickstart.js
const fs = require("fs");

const apiKey = process.env.SMALLEST_API_KEY;
if (!apiKey) {
  console.error("Error: set SMALLEST_API_KEY environment variable");
  console.error("Get your key at https://app.smallest.ai/dashboard/settings/apikeys");
  process.exit(1);
}

fetch("https://api.smallest.ai/waves/v1/lightning-v3.1/get_speech", {
  method: "POST",
  headers: { Authorization: `Bearer ${apiKey}`, "Content-Type": "application/json" },
  body: JSON.stringify({ text: "Hello! Welcome to Smallest AI. This is your first text-to-speech generation.", voice_id: "sophia", sample_rate: 24000, output_format: "wav" }),
})
  .then((r) => {
    if (!r.ok) return r.text().then((t) => { throw new Error(`API error (${r.status}): ${t}`); });
    return r.arrayBuffer();
  })
  .then((buf) => {
    fs.writeFileSync("output.wav", Buffer.from(buf));
    console.log(`Done! Saved output.wav (${Buffer.from(buf).length.toLocaleString()} bytes)`);
  })
  .catch((err) => { console.error(err.message); process.exit(1); });
